browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -233,12 +233,11 @@ void BraveShieldsWebContentsObserver::DispatchBlockedEventForWebContents(
 #endif
   if (base::FeatureList::IsEnabled(
           brave_shields::features::kBraveShieldsPanelV2)) {
-#if !BUILDFLAG(IS_ANDROID)
     if (!web_contents)
       return;
-    brave_shields::BraveShieldsDataController::FromWebContents(web_contents)
-        ->HandleItemBlocked(block_type, subresource);
-#endif
+    auto* shields_data_ctrlr = brave_shields::BraveShieldsDataController::FromWebContents(web_contents);
+    DCHECK(shields_data_ctrlr);
```

Comment:
@bridiver wanted this DCHECK

Reply:
@simonhong a WebContentsUserData should never be destroyed before the webcontents. The only time this could be null is if the webcontents is not associated with a tab, but that seems like it would be a bug in the networking code and hence the DCHECK here

Reply:
Ok, I was worried some WebContents that not associated with tab might be reached here.

Reply:
@bridiver btw, curious how networking code is tied with tab's webcontents. can you point out where it is?